### ISSUE TITLE
Add-on-hover isn't saved

### ISSUE THREAD

**Issue by vphantom** [tomitm/lists#8]
Hi!

First, thank you for this extension. It's amazing how buried lists are in Twitter's default UI...  That said, on the 23rd I noticed that even though checkboxes get momentarily checked in the hover user info, they aren't actually added to lists anymore.  If I go to the user's page and use the cog button, then the "add/remove from lists" feature built into Twitter, I can still add them, however.

I guess my first question is: can you reproduce this, or is my situation unique?

Thank you for your time.

**Comment by tomitm** [comment]
Hi @vphantom! Thanks for submitting an issue, I wasn't aware this was happening. 😞

I just tried to reproduce really quickly, but it looks like it's not a problem for me, so it might just be happening in certain cases.

I'll dig around when I get a chance, but could you give me a bit more info so I can narrow down the issue:

- Is this always happening everywhere?
- What page(s) does this happen on? (main feed, list feed, search results page, etc.)
- What was the account you tried adding?
- Was the list public or private?
- [optional, if you're able to] Can you provide me with any info from devtools, like a log, or failed network request? (bonus points for the `POST /i/123/lists/123/members` response)

A screenshot or gif/movie would probably be best to answer most of these questions. Feel free to blur things out, if you'd like.

The interesting thing is the checkbox only gets checked after a successful add request completes. Though, "success" isn't actually constrained much right now, so it's entirely possible I'm accepting a false-positive in your case. Also interesting is that I'm mimicking the same request as that 'add/remove from lists' feature you say works, but it's possible something changed there too. 🤔

**Comment by vphantom** [comment]
I'll run some experiments with the developer tools activated and report back. So far it is consistent: wherever I raise the little user pop-up and click a list to add them in it, after a (presumably network) delay the box is checked, but next time around it won't be, and indeed the user isn't in the list itself.

So I'm suspecting Twitter may have changed its REST API slightly and is perhaps rolling it out to some users before others (like YouTube loves to do).

**Comment by vphantom** [comment]
Huh, checking the box sends a POST to `https://twitter.com/i/599876632/lists/813739900952596480/members` which results in a `403 Forbidden` and the checkbox neverthess appears momentarily checked.

The body of the response: `{"message":"Sorry! We did something wrong."}`

**Comment by tomitm** [comment]
I'm using the undocumented API they use for their website, so I wouldn't be surprised if it changed.

Since you're seeing a 403, that sounds like `authenticity_token` might not be getting sent in the POST form data, or if it is, Twitter might've changed something else (additional header, etc).

I'll see what I can find out, but you're right, it might be a rollout I'm not seeing yet. Feel free to submit a PR if you can track it down in the meantime. 🙂

**Comment by vphantom** [comment]
I found the culprit in the network I/O. I'm keeping my notes below as-is from my investigation in case it helps, but here's the spoiler: the user pop-up defines `_method=DELETE` in the post request when I _check_ the checkbox, whereas it should only define it when _unchecking_ the checkbox.

From what I see in `src/profile-card.js` line 49, you're making it properly conditional. The condition gets minified to `a.checked&&(l._method="DELETE")` which seems OK.  So, while I found the culprit, I don't understand why it would appear that in Chrome 55.0.2883.87 (64-bit) I seem to be getting `inputEl=true` when the box is unchecked.


## Using Twitter's UI

### Adding:

#### General
```
Request URL:https://twitter.com/i/106037462/lists/813739900952596480/members
Request Method:POST
Status Code:200
```

#### Form Data
```
authenticity_token:0fda6cc3b4f3c6fdaa51f032xxxxxxxxxxxxxxxx
```

#### Response
```
{}
```

### Removing:

#### General
```
Request URL:https://twitter.com/i/106037462/lists/813739900952596480/members
Request Method:POST
Status Code:200
```

#### Form Data
```
_method:DELETE
authenticity_token:0fda6cc3b4f3c6fdaa51f032xxxxxxxxxxxxxxxx
```

## Using pop-up UI

#### General
```
Request URL:https://twitter.com/i/106150812/lists/813739900952596480/members
Request Method:POST
Status Code:403
```

#### Form Data
```
_method:DELETE
authenticity_token:0fda6cc3b4f3c6fdaa51f032xxxxxxxxxxxxxxxx
```

#### Response
```
{"message":"Sorry! We did something wrong."}
```

**Comment by tomitm** [comment]
Yep, just confirmed that's the response you get when you try to delete a membership that doesn't exist.

Thanks for narrowing that down! You're right that it would have something to do with [the checkbox](https://github.com/tomitm/lists/blob/master/src/profile-card.js#L48-L50).

Aand I just reproduced it. I'm going to go ahead and guess you're clicking directly on the checkbox, which is what causes the issue.

Apparently it never occured to me to test that before, as I've been clicking on the rest of the row this whole time (which works). This explains why `inputEl` is already checked and is triggering `DELETE`. I think past me figured `e.preventDefault()` would've suppressed the user-initiated checkmark, but I clearly didn't test that well enough.

I'll try to push out a fix soon-ish. Thanks for catching that!

**Comment by tomitm** [comment]
Quick fix is in the `checkbox-fix` branch. 🔧

If you'd like to try it out, you can [build from source](https://github.com/tomitm/lists/wiki/Building-from-source). Otherwise, I'd recommend clicking on the label/row instead of the checkbox for now.

Will release in the next day or so, just want to tidy up a bit first (e.g., better error handling/reporting).

In case you're curious about the bug: have a look at a [reduced test case](https://jsfiddle.net/pre4yq4s/).

**Comment by vphantom** [comment]
Clean fix! This also helps explain why I could add people to lists in the past. Thank you for looking into it!

**Comment by tomitm** [comment]
@vphantom Finally got around to pushing this out, sorry for the delay!

Chrome should be auto-updating you in the next 0-24 hours. 🎈
